grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java

```diff
@@ -75,27 +88,35 @@ static RemoteGraknTx create(RemoteGraknSession session, GraknTxType txType) {
 
     @Override
     public EntityType putEntityType(Label label) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, ENTITY);
     }
 
     @Override
     public <V> AttributeType<V> putAttributeType(Label label, AttributeType.DataType<V> dataType) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, ATTRIBUTE, var("x").datatype(dataType));
     }
 
     @Override
     public Rule putRule(Label label, Pattern when, Pattern then) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, RULE, var("x").when(when).then(then));
```

Comment:
Maybe we can extract `var("x")` into a constant, since otherwise it looks a bit magic.

Or go full functional and pass a lambda 😛 
```
return putSchemaConcept(label, RULE, var -> var.when(when).then(then));
```

Reply:
I actually like the functional approach